app/src/main/java/org/fossasia/openevent/api/APIClient.java

```diff
@@ -37,6 +48,15 @@ public APIClient() {
                 .setConverter(new GsonConverter(gson))
                 .setEndpoint(Urls.BASE_GET_URL_ALT)
                 .setLogLevel(RestAdapter.LogLevel.FULL)
+                .setErrorHandler(new ErrorHandler() {
+                    @Override
+                    public Throwable handleError(RetrofitError cause) {
+                        ErrorDialogEvent errEvent = new ErrorDialogEvent(null, null);
+                        errEvent.ErrorHandlerFunction(cause, errEvent);
+                        OpenEventApp.postEventOnUIThread(new ErrorDialogEvent(errEvent.ErrorType, errEvent.ErrorDesc));
+                        return new Exception(cause);
```

Comment:
Yeah, the error dialogs only differ in the type of error and the description. Check out HandleErrorDialogEvent. The constructor takes in two strings.